packages/block-library/src/index.js

```diff
@@ -176,56 +137,70 @@ const getAllBlocks = () =>
 		video,
 
 		// theme blocks
-		navigation,
-		navigationLink,
-		navigationSubmenu,
 		siteLogo,
 		siteTitle,
 		siteTagline,
-		query,
 		templatePart,
 		avatar,
 		postTitle,
 		postExcerpt,
 		postFeaturedImage,
-		postContent,
 		postAuthor,
 		postAuthorName,
 		postComment,
 		postCommentsCount,
 		postCommentsLink,
 		postDate,
 		postTerms,
-		postNavigationLink,
-		postTemplate,
 		postTimeToRead,
-		queryPagination,
-		queryPaginationNext,
-		queryPaginationNumbers,
-		queryPaginationPrevious,
-		queryNoResults,
+		postNavigationLink,
 		readMore,
 		comments,
 		commentAuthorName,
 		commentContent,
 		commentDate,
 		commentEditLink,
 		commentReplyLink,
-		commentTemplate,
 		commentsTitle,
-		commentsPagination,
-		commentsPaginationNext,
-		commentsPaginationNumbers,
-		commentsPaginationPrevious,
 		postCommentsForm,
 		tableOfContents,
-		homeLink,
 		logInOut,
 		termDescription,
 		queryTitle,
 		postAuthorBiography,
 	].filter( Boolean );
 
+export const getAsyncBlocks = () => [
+	[ 'button', 'buttons' ],
+	[ 'column', 'columns' ],
+	[ 'image', 'gallery' ],
+	[
+		'post-template',
+		'query-pagination-previous',
+		'query-pagination-numbers',
+		'query-pagination-next',
+		'query-pagination',
+		'query-no-results',
+		'query',
+	],
+	[
+		'comments-pagination-previous',
+		'comments-pagination-next',
+		'comments-pagination-numbers',
+		'comments-pagination',
+		'comment-template',
+	],
+	[ 'home-link', 'navigation-link', 'navigation-submenu', 'navigation' ],
+	[ 'page-list', 'page-list-item' ],
+	[ 'social-link', 'social-links' ],
+	[ 'nextpage', 'post-content' ],
+];
+
+export const asyncLoadBlock = async ( blockType ) => {
+	const { init } = await import( './' + blockType + '/index.js' );
+	init();
```

Comment:
There are entry points for every core block that init them automatically:
```suggestion
	await import( './' + blockType + '/init.js' );
```